packages/fxa-auth-server/test/remote/account_reset_tests.js

```diff
@@ -21,239 +22,194 @@ describe('remote account reset', function() {
       password,
       server.mailbox
-    )
-      .then(x => {
-        client = x;
-      })
-      .then(() => {
-        return client.forgotPassword();
-      })
-      .then(() => {
-        return server.mailbox.waitForCode(email);
-      })
-      .then(code => {
-        assert.throws(() => {
-          client.resetPassword(newPassword);
-        });
-        return resetPassword(client, code, newPassword);
-      })
-      .then(response => {
-        assert.ok(response.sessionToken, 'session token is in response');
-        assert(
-          !response.keyFetchToken,
-          'keyFetchToken token is not in response'
-        );
-        assert.equal(response.verified, true, 'verified is true');
-      });
+    );
+
+    await client.forgotPassword();
+    const code = await server.mailbox.waitForCode(email);
+    assert.throws(() => {
+      client.resetPassword(newPassword);
+    });
+    const response = await resetPassword(client, code, newPassword);
+    assert.ok(response.sessionToken, 'session token is in response');
+    assert(!response.keyFetchToken, 'keyFetchToken token is not in response');
+    assert.equal(response.verified, true, 'verified is true');
   });
 
-  it('account reset deletes tokens', () => {
+  it('account reset deletes tokens', async () => {
     const email = server.uniqueEmail();
     const password = 'allyourbasearebelongtous';
     const newPassword = 'ez';
-    let client = null;
-    let originalCode = null;
     const opts = {
       keys: true,
     };
-    return Client.createAndVerify(
+
+    const client = await Client.createAndVerify(
       config.publicUrl,
       email,
       password,
       server.mailbox,
       opts
-    )
-      .then(x => {
-        client = x;
+    );
+
+    await client.forgotPassword();
+    // Stash original reset code then attempt to use it after another reset
+    const originalCode = await server.mailbox.waitForCode(email);
+
+    await client.forgotPassword();
+    const code = await server.mailbox.waitForCode(email);
+    assert.throws(() => client.resetPassword(newPassword));
+    await resetPassword(client, code, newPassword, undefined, opts);
+
+    const emailData = await server.mailbox.waitForEmail(email);
+    const templateName = emailData.headers['x-template-name'];
+    assert.equal(templateName, 'passwordResetEmail');
+
+    try {
+      await resetPassword(client, originalCode, newPassword, undefined, opts);
+      assert.fail('Should not have succeeded password reset');
+    } catch (err) {
+      // Ensure that password reset fails with unknown token error codes
+      assert.equal(err.code, 401);
+      assert.equal(err.errno, 110);
+    }
+  });
 
-        return client.forgotPassword();
-      })
-      .then(() => server.mailbox.waitForCode(email))
-      .then(code => {
-        // Stash original reset code then attempt to use it after another reset
-        originalCode = code;
+  it('account reset updates keysChangedAt', async () => {
```

Comment:
This is the only new test in this file, the others were just refactored to async/await.